src/main/java/io/confluent/kafkarest/ConsumerReadTask.java

```diff
@@ -143,7 +144,9 @@ public boolean doPartialRead() {
           started + parent.getConfig().getInt(KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG);
       waitExpiration = Math.min(backoffExpiration, requestExpiration);
 
-      if (elapsed >= requestTimeoutMs || bytesConsumed >= maxResponseBytes) {
+      // Including the rough message size here ensures processing finishes if the next
+      // message exceeds the maxResponseBytes
+      if (elapsed >= requestTimeoutMs || bytesConsumed + roughMsgSize >= maxResponseBytes) {
```

Comment:
Include the check on the roughMsgSize here so that we break out if it has been exceeded.